### ISSUE TITLE
Code quality: Define one lexer in *one* file

### ISSUE THREAD

**Issue by rugk** [pygments/pygments#1522]
(First contributor experience here.)

My fix for https://github.com/pygments/pygments/issues/1520 was heavily complicated by the fact that I first had to find the actual file for the HTTP lexer and it did not had "HTTP" in it's name.
Thus, the GitHub and locally file search did not work to find the lexer. Also GitHub's full-text search was not helpful, because the string `http` is of course used everywhere and in the actual lexer file not really present, _or_ at least not in a "whole-word-only" search that GItHub AFAIK does. (i.e. `HttpLexer` could not be found)
Only searching for `POST` as a thing had had to be defined there was successful. :roll_eyes: 
IMHO, that was too cumbersome.

Please, by all means, there is no need to put multiple lexers into a single file.
One file for one lexer would make it _much_ easier to find and thsu to maintain. There is no need to limit the amounts of files you use.
_If_ you want to group them, just use folders.

May also be applied to formatters or so. Did not look into whether they are affected by this problem as well.

**Comment by Anteru** [comment]
@birkenfeld I added an explicit statement to encourage separate modules for lexers via 9ef94104d928bb0c26da26fb711b0490ec401123

**Comment by rugk** [comment]
> My suggestion would be to search here: https://pygments.org/docs/lexers/ -- would that have worked for you?

Yeah, but did not find that website. Only looked at [this](https://pygments.org/languages/) and yeah… now that I know your tech-term lexers it makes sense you duplicate this, but hmm… yeah, the one doc is more technical, but _technically_ (pun not intended) they still document the same: the list of languages/lexers you have.

So it did not even came to my mind to search the docs for another page, the one I have did not help.

Now that I rethink I actually looked at https://pygments.org/docs/lexerdevelopment/, but alos saw just things for creating a new lexer.

TL;DR: I guess some links would help here.

But I also have to say I was in a hurry, so 
> Using slightly more specific search terms (I tried "http lexer", for example), gets you to the correct class quickly.

did not even occur to me. Also I did not know this works on GitHub (I thought with spaces it just does an `OR` search… dunno)
